### PR TITLE
Stabilization Build

### PR DESCRIPTION
**Pull request by Jered**
## Fixes
- fixes initial momentum move (no delay to move without scale)
- fixes cleanup of impetus
- decouples zoom state from animating state
- accepts startingZoom option
- cleanup

## Known issues
- double tap is janky (don't use for now)
- still need bounce back and resistance beyond min/max scale
- not using momentum is painful, should be improved or ripped out and only have inertial movement

### CODE REVIEW

**Comment by Jered** [comment]
@thomasbruketta: fixed the issues we were seeing. Should be good to go with the following notes:

- doubletap still broken but we're not using it currently
- bounceback and resistance at min/max scales not implemented
- destroy method not working with hotloader but may be in prod too. JS cleanup error that may need to be handled in the composing component.

```
[Error] TypeError: null is not an object (evaluating 'this.pinchZoomImage.destroy')
	__reset__REACT_HOT_LOADER__ (Anonymous Script 14 (line 190))
	(anonymous function) (Anonymous Script 14 (line 174))
```

**Comment by Jered** [comment]
Update: Console error was fixed in project. Not an issue of this component.

File: pinch-zoom-canvas.js [hunk 1]
```diff
@@ -1,468 +1,528 @@
...
+    this.fullScreen          = options.fullScreen || false
+
+    // Init
+    this.position = {
+      x: 0,
+      y: 0
+    }
+    this.scale = {
+      x: 1,
+      y: 1
```
Comment (thomasbruketta) [inline pinch-zoom-canvas.js]: missing trailing comma

File: pinch-zoom-canvas.js [hunk 2]
```diff
@@ -1,468 +1,528 @@
...
+    }
+    this.initPosition = {
+      x: 0,
+      y: 0
```
Comment (thomasbruketta) [inline pinch-zoom-canvas.js]: missing trailing comma

File: pinch-zoom-canvas.js [hunk 3]
```diff
@@ -1,468 +1,528 @@
...
+    }
+    this.offset = {
+      x: 0,
+      y: 0
```
Comment (thomasbruketta) [inline pinch-zoom-canvas.js]: comma

File: pinch-zoom-canvas.js [hunk 4]
```diff
@@ -1,468 +1,528 @@
...
+    this.onZoom              = options.onZoom // Callback on zoom
+    this.initResizeProperty  = null
+    this.threshold           = options.threshold || 40
+    this.startingZoom        = options.startingZoom || 1
+    this.fullScreen          = options.fullScreen || false
+
+    // Init
+    this.position = {
+      x: 0,
+      y: 0
```
Comment (thomasbruketta) [inline pinch-zoom-canvas.js]: comma

File: pinch-zoom-canvas.js [hunk 5]
```diff
@@ -1,468 +1,528 @@
...
+        this.offset.y = canvasBox.top + window.scrollY
+      }
+
+      return this
+    },
+
+    /**
+     * handles zooming in and out
+     */
+    zoom: function (zoom, touchX, touchY) {
```
Comment (thomasbruketta) [inline pinch-zoom-canvas.js:177]: Denote private method using `_zoom` to match other private methods below

File: pinch-zoom-canvas.js [hunk 6]
```diff
@@ -1,468 +1,528 @@
...
+          this.position.x = relativeX
+          this.position.y = relativeY
+        }
+      }
+
+      this.lastX = relativeX
+      this.lastY = relativeY
+    },
+
+    isZoomed: function () {
```
Comment (thomasbruketta) [inline pinch-zoom-canvas.js:293]: Private method?

File: pinch-zoom-canvas.js [hunk 7]
```diff
@@ -1,468 +1,528 @@
...
+      this.position.y += pY * deltaHeight
+
+      // onZoom callback
+      if (this.onZoom) {
+        this.onZoom(newScale, this.zoomed)
+      }
+
+    },
+
+    move: function (relativeX, relativeY) {
```
Comment (thomasbruketta) [inline pinch-zoom-canvas.js:224]: `_move`?

File: pinch-zoom-canvas.js (hunk 5 above)
Comment (Jered) [reply pinch-zoom-canvas.js:177]: this should remain public as you can call it externally to force move things outside of user input

File: pinch-zoom-canvas.js (hunk 7 above)
Comment (Jered) [reply pinch-zoom-canvas.js:224]: this should remain public as you can call it externally to force move things outside of user input

File: pinch-zoom-canvas.js (hunk 6 above)
Comment (Jered) [reply pinch-zoom-canvas.js:293]: isZoomed is a public getter

**Pull request merged**